### PR TITLE
tricore: lift remaining floating point instructions

### LINKED ISSUE #4115: tricore: unlifted floating point instructions
- [x] `FTOI`
- [x] `FTOQ31Z`
- [x] `FTOQ31`
- [x] `FTOUZ`
- [x] `FTOU`
- [x] `HPTOF`
- [x] `ITOF`
- [x] `Q31TOF`
- [x] `DIV_F`
- [x] `ADD_F`
- [x] `MADD_F`
- [x] `MSUB_F`
- [x] `SUB_F`
- [x] `MUL_F`
- [x] `QSEED_F`
- [x] `UTOF`

### PR DESCRIPTION
**Pull request by b1llow**
# SQUASH ME

**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

closes #4115

### CODE REVIEW

File: librz/arch/isa/tricore/tricore_il_fp.inc [hunk 1]
```diff
@@ -0,0 +1,591 @@
+// SPDX-FileCopyrightText: 2024 billow <billow.fun@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+static const ut32 HP_MAX_VALUE = 65504;
```
Comment (XVilka) [inline librz/arch/isa/tricore/tricore_il_fp.inc:5]: Please add a comment what this means, not everyone knows what `HP` is, I guess it's the "half-precision float", right?

File: librz/arch/isa/tricore/tricore_il_fp.inc [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+	return ITE(IS_FNAN(VARL(k)),
+		U32(0),
+		ITE(FGT(VARL(k), F64(max)),
+			U32(max),
+			ITE(FLT(VARL(k), F64(min)),
+				U32(min),
+				x)));
+}
+
+#define Q31_MAX q_real(0x7FFFFFFF)
```
Comment (XVilka) [inline librz/arch/isa/tricore/tricore_il_fp.inc:99]: Same

File: librz/include/rz_il/rz_il_opbuilder_begin.h [hunk 3]
```diff
@@ -99,12 +103,12 @@
 #define AND(x, y) rz_il_op_new_bool_and(x, y)
 #define OR(x, y)  rz_il_op_new_bool_or(x, y)
 
-#define FNEQ(flx, fly) rz_il_op_new_fneq(flx, fly)
-#define FEQ(flx, fly)  rz_il_op_new_feq(flx, fly)
-#define FLT(flx, fly)  rz_il_op_new_flt(flx, fly)
-#define FLE(flx, fly)  rz_il_op_new_fle(flx, fly)
-#define FGT(flx, fly)  rz_il_op_new_fgt(flx, fly)
-#define FGE(flx, fly)  rz_il_op_new_fge(flx, fly)
+#define FNE(flx, fly) rz_il_op_new_fneq(flx, fly)
```
Comment (XVilka) [inline librz/include/rz_il/rz_il_opbuilder_begin.h:106]: Do you think renaming `FNEQ()` helper macro to `FNE()` really necessary? @thestr4ng3r @DMaroo @Rot127 @wargio what do you think?

File: librz/arch/p/analysis/analysis_tricore_cs.c [hunk 4]
```diff
@@ -241,7 +239,12 @@ static char *tricore_reg_profile(RzAnalysis *_) {
 		"drx	FPU_TRAP_OPC	.32	812	0\n"
 		"drx	FPU_TRAP_SRC1	.32	816	0\n"
 		"drx	FPU_TRAP_SRC2	.32	820	0\n"
-		"drx	FPU_TRAP_SRC3	.32	824	0\n";
+		"drx	FPU_TRAP_SRC3	.32	824	0\n"
+		"drx set_FI .1 900 0\n"
```
Comment (XVilka) [inline librz/arch/p/analysis/analysis_tricore_cs.c]: Please indent it to match other registers.

**Review by XVilka — CHANGES_REQUESTED** [review]
Amazing job, as usual! Please add a floating-point Tricore "emulateme" test

File: librz/include/rz_il/rz_il_opbuilder_begin.h (hunk 3 above)
Comment (wargio) [reply librz/include/rz_il/rz_il_opbuilder_begin.h:106]: ok with that but these changes also needs to be done in `rz_il_opbuilder_end.h`

Comment (DMaroo) [reply librz/include/rz_il/rz_il_opbuilder_begin.h:106]: I really don't have a strong preference. But `FNE` does fit in nicer with all the other 3 letter floating-point comparison macros.

File: librz/include/rz_il/rz_il_opbuilder_end.h [hunk 5]
```diff
@@ -55,7 +55,7 @@
 #undef FMOD
 #undef FPOW
 #undef FMAD
-#undef FNEQ
+#undef FEQ
```
Comment (XVilka) [inline librz/include/rz_il/rz_il_opbuilder_end.h]: Should be this `FNE` instead?

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM with one minor problem

File: librz/include/rz_il/rz_il_opbuilder_end.h (hunk 5 above)
Comment (b1llow) [reply librz/include/rz_il/rz_il_opbuilder_end.h]: Yes, my mistake.

**Review by XVilka — APPROVED** [review]

**Pull request merged**